cluster-autoscaler/processors/nodegroups/nodegroup_pool_list_processor.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodegroups
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// NodeGroupPoolListProcessor processes list of pools of NodeGroup considered in scale-up.
+// This list of pools can serve as priority. Pool which comes first has higher priorty.
+type NodeGroupPoolListProcessor interface {
+	Process(context *context.AutoscalingContext, nodeGroupPoolList [][]cloudprovider.NodeGroup,
```

Comment:
nit: I feel like taking `nodeGroupPools []cloudprovider.NodeGroup` instead of `nodeGroupPoolList [][]cloudprovider.NodeGroup` would be a bit cleaner (same arguments as `NodeGroupListProcessor`), but don't have a strong opinion on that.

Reply:
Yes, I changed that in the previous commit.